helpers/controllers/userInfoController.ts

```diff
@@ -35,21 +35,60 @@ export const userInfo = async (_parent: void, args: UserInfoQueryVariables) => {
     prisma.star.findMany({
       where: {
         mentorId: user.id
+      },
+      include: {
+        student: {
+          select: {
+            username: true,
+            name: true
+          }
+        },
+        lesson: {
+          select: {
+            title: true,
+            order: true
+          }
+        }
+      }
+    }),
+    prisma.star.findMany({
```

Comment:
I think this query to find the stars given by the user should remain separated from the profile query since this data is not used in the profile page.

What you could do is call this in the session resolver and make this query there where it's needed.

Reply:
This definitely belongs on the session resolver. I was thinking since session resolver is just a userInfo resolver with a username passed in, could just put this here and then session resolver just calls userInfo. Would that be an antipattern?

Reply:
Yeah I get you, but the way you did it will make this additional query even for the profile resolver, which does nothing with this data. The way I suggested would be to call this in the session resolver, and then just add this star given query there where it's needed. Something like this for example:

```js
const session = async (req) => {
  const { username } = req.user
  const [session, starsGiven] = await Promise.all([
    userInfo(username),
    prisma.star.findMany({ ... }) // get the stars given by this user
  ])
  // do the additional mapping
  return {
    user,
    lessonStatus,
    submissions
  }
}
```

Reply:
ah, yeah that's much better